sdk/tests/core/mock_az_mqtt.c

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Contains the az_mqtt.h mock implementation.
+ *
+ */
+
+#include <azure/core/az_mqtt.h>
+#include <azure/core/az_mqtt_config.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_span.h>
+#include <azure/core/internal/az_log_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/internal/az_result_internal.h>
+#include <azure/core/internal/az_span_internal.h>
+
+#include <mosquitto.h>
+#include <pthread.h>
+#include <stdlib.h>
+
+#include <azure/core/_az_cfg.h>
+
+#ifdef _az_MOCK_ENABLED
+
+AZ_NODISCARD az_result __wrap_az_mqtt_init(az_mqtt* mqtt, az_mqtt_options const* options);
+AZ_NODISCARD az_result __wrap_az_mqtt_init(az_mqtt* mqtt, az_mqtt_options const* options)
+{
+  (void)mqtt;
+  (void)options;
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result
+__wrap_az_mqtt_outbound_connect(az_mqtt* mqtt, az_mqtt_connect_data* connect_data);
```

Comment:
I'm not sure I understand why this is needed: can't we simply provide an az_mqtt.h implementation and compile that into the UT instead of using GCC's mocks?

Reply:
Resolving as part of our discussion regarding gcc mocks.